### ISSUE TITLE
Deploys of this template only respond with 404

### ISSUE THREAD

**Issue by jesper-bylund** [vercel/platforms#324]
Following the official guide (https://vercel.com/blog/platforms-starter-kit) and deploying the template (https://vercel.com/templates/next.js/platforms-starter-kit) only starts a service that responds 404 to each request _without any errors or logs to debug_.

It works locally. Just not on vercel as a host.

There are many issues and discussions, all unanswered for weeks or months.
Is this project abandoned? Is it supposed to work? How can we debug the 404 error? 🤔

**Comment by jesper-bylund** [comment]
ping @rauchg

**Comment by jesper-bylund** [comment]
I hope my tone isn't harsh here. I'm excited about this template and would love to use it, which is why the apparent wall of silence is frustrating. 🙏

**Comment by jesper-bylund** [comment]
ping @steven-tey, who seems to have been the primary contributor to this repo?

**Comment by steven-tey** [comment]
Just merged #323 which should fix this issue! Anyone still facing this problem?

**Comment by jesper-bylund** [comment]
@steven-tey that didn't solve anything. 🤔 Literally no difference. How did you test that? My deployed version still only shows 404. both on `root.tld` and `app.root.tld`

**Comment by jesper-bylund** [comment]
So thanks to your update I realised I could use middleware.ts to debug what was happening.
It turns out that `NEXT_PUBLIC_ROOT_DOMAIN` has to be an exact match with with the variable hostname.

This caused issues for me because my NEXT_PUBLIC_ROOT_DOMAIN was using the protocol prefix https:// 😅

This is a fragile setup. There's quite a few tiny things that you need to get exactly right about the wildcard domain and the variable setup that could be better documented.

TlDr;
My `NEXT_PUBLIC_ROOT_DOMAIN` was `https://note.fish` and my `hostname` was `note.fish`. This caused infinite 404s.

**Comment by Neosprings** [comment]
I guess I'm running into this issue as well. Any better documentation on how to get past the 404 mainpage?
